src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,46 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.*;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        BulldozerProducer bulldozerProducer = new BulldozerProducer();
+        TruckProducer truckProducer = new TruckProducer();
+        ExcavatorProducer excavatorProducer = new ExcavatorProducer();
```

Comment:
Creating a new instance of the producer classes every time `getAll()` is called can lead to performance issues. Consider making these producers as class fields and initialize them only once in the constructor or directly in the field declaration.